From f6d4b020587a84e2152366a287d468089b92cbcb Mon Sep 17 00:00:00 2001
From: Alexandros Frantzis <alexandros.frantzis@collabora.com>
Date: Fri, 22 Aug 2025 16:49:58 +0300
Subject: [PATCH 345/346] winewayland: Mark only windows with per-pixel alpha
 as layered.

This avoids using buffers with an alpha channel for windows that
use SetLayeredWindowAttributes() and don't really utilize
per-pixel alpha.

Fixes: 6f8f5fe7bf989ba1429bc42114091e8416ad1a7c
Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=58504
---
 dlls/winewayland.drv/window_surface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winewayland.drv/window_surface.c b/dlls/winewayland.drv/window_surface.c
index 61bede22c5c..eb232bae62a 100644
--- a/dlls/winewayland.drv/window_surface.c
+++ b/dlls/winewayland.drv/window_surface.c
@@ -524,7 +524,7 @@ BOOL WAYLAND_CreateWindowSurface(HWND hwnd, BOOL layered, const RECT *surface_re
     if (previous) window_surface_release(previous);
 
     if (layered) data->layered_attribs_set = TRUE;
-    *surface = wayland_window_surface_create(data->hwnd, surface_rect, data->layered_attribs_set);
+    *surface = wayland_window_surface_create(data->hwnd, surface_rect, layered);
 
     wayland_win_data_release(data);
     return TRUE;
-- 
2.51.0

